api/mappers/engine.py

```diff
@@ -0,0 +1,404 @@
+    from features.models import Feature, FeatureSegment, FeatureState
+    from features.multivariate.models import (
+        MultivariateFeatureOption,
+        MultivariateFeatureStateValue,
+    )
+    from integrations.common.models import EnvironmentIntegrationModel
+    from integrations.dynatrace.models import DynatraceConfiguration
+    from integrations.webhook.models import WebhookConfiguration
+    from organisations.models import Organisation
+    from projects.models import Project
+    from segments.models import Segment, SegmentRule
+
+
+__all__ = (
+    "map_environment_api_key_to_engine",
+    "map_environment_to_engine",
+    "map_feature_to_engine",
+    "map_identity_to_engine",
+    "map_mv_option_to_engine",
+)
+
+
+def map_segment_rule_to_engine(
+    segment_rule: "SegmentRule",
+) -> SegmentRuleModel:
+    segment_sub_rules = segment_rule.rules.all()
+    conditions = segment_rule.conditions.all()
+
+    return SegmentRuleModel(
+        type=segment_rule.type,
+        rules=[
+            map_segment_rule_to_engine(segment_sub_rule)
+            for segment_sub_rule in segment_sub_rules
+        ],
+        conditions=[
+            SegmentConditionModel(
+                operator=condition.operator,
+                value=condition.value,
+                property_=condition.property,
+            )
+            for condition in conditions
+        ],
+    )
+
+
+def map_integration_to_engine(
+    integration: Optional["EnvironmentIntegrationModel"],
+) -> Optional[IntegrationModel]:
+    if not integration:
+        return None
+    return IntegrationModel(
+        api_key=integration.api_key,
+        base_url=integration.base_url,
+    )
+
+
+def map_dynatrace_integation_to_engine(
+    integration: Optional["DynatraceConfiguration"],
+) -> Optional[IntegrationModel]:
+    if not integration:
+        return None
+    return IntegrationModel(
+        api_key=integration.api_key,
+        base_url=integration.base_url,
+        entity_selector=integration.entity_selector,
+    )
+
+
+def map_webhook_config_to_engine(
+    webhook_config: Optional["WebhookConfiguration"],
+) -> Optional[WebhookModel]:
+    if not webhook_config:
+        return None
+    return WebhookModel(
+        url=webhook_config.url,
+        secret=webhook_config.secret,
+    )
+
+
+def map_feature_state_to_engine(
+    feature_state: "FeatureState",
+    mv_fs_values: Iterable["MultivariateFeatureStateValue"],
+) -> FeatureStateModel:
+    feature = feature_state.feature
+    feature_segment: Optional["FeatureSegment"] = feature_state.feature_segment
+
+    if feature_segment:
+        feature_segment_model = FeatureSegmentModel(
+            priority=feature_segment.priority,
+        )
+    else:
+        feature_segment_model = None
+
+    return FeatureStateModel(
+        enabled=feature_state.enabled,
+        django_id=feature_state.pk,
+        feature_state_value=feature_state.get_feature_state_value(),
```

Comment:
Do we need to consider the identity hash key here?

Reply:
This mapper is only used for environment models, so no.

Reply:
I don't think that's true is it? 

https://github.com/Flagsmith/flagsmith/pull/2290/files/3506c14a0e2272589416669eefd8e021bb025021#diff-967f7cb2f63496c9174f36c8a54cdbaca392764a77f891ae4b27efcc6bf612eaR349

Reply:
Yes, this is actually not true — this mapper is also used for identities.

However, this implementation completely mirrors the behaviour of current engine's schemas (see https://github.com/Flagsmith/flagsmith-engine/blob/625a22cb29ed42c5a17b62889d10cf276f55476b/flag_engine/api/schemas.py#L55-L56). This is probably intentional because multivariate feature states are meant to be resolved later in the API business logic.

Please correct me if I didn't get it right.

Reply:
Yep, that makes sense!